src/workspace/workspacetypes.h

```diff
@@ -61,6 +61,13 @@ struct ToolbarData : public AbstractData
 
 using ToolbarDataPtr = std::shared_ptr<ToolbarData>;
 
+struct UiArrangmentData : public AbstractData
+{
+    std::map<std::string /*key*/, Val> values;
```

Comment:
I think this is a too abstract way to store a window state. Better to store state directly in QByteArray. Maybe it's better to create a special struct:

struct PageState {
std::string pageName;
QByteArray state;
}

Reply:
there may be other data in the future

Reply:
If we want to store a new property for a window then we can just add a new member to PageState. Val is a too abstract way to store data in this case, because you only need to store an array of bytes. Val can store anything and it can be confusing

Reply:
If you look at the implementation of UiConfiguration::uiValue, then you will understand that UiArrangment is similar to settings. Key - value. We don't know what keys are stored and what types of values are there. And it's better not to complicate with this knowledge